### ISSUE TITLE
Android controllers not working with hat dpad controls

### ISSUE THREAD

**Issue by andybotting** [libretro/retroarch-joypad-autoconfig#226]
Looks like autodetect is working OK, except I've found on Android that the dpad hat controls don't work. 

I've tested only an OUYA controller so far and found once I remapped:

```
input_up_btn = "h0up"
input_down_btn = "h0down"
input_left_btn = "h0left"
input_right_btn = "h0right"
```

to 

```
input_up_btn = "19"
input_down_btn = "20"
input_left_btn = "21"
input_right_btn = "22"
```

everything works great.

I haven't tested anything else yet, but I do have an old orig XBox controller + USB adapters and a few other random controllers lying around if you're looking for more comprehensive testing.

Thanks

**Comment by hizzlekizzle** [comment]
ugh, I hate when stuff like that changes for no good reason... Feel free to submit PRs to fix any that you come across and can test.

**Comment by fr500** [comment]
This is probably device specific though

**Comment by andybotting** [comment]
Yeah. Just tested my old XBox controller, and it does work with the hats.

For now, will push up a pull request for the OUYA controller. Have a Xiaomi controller coming in the post next week so will make sure I map that one too.

With my XBox controller and a dodgy USB controller, they have no config so no keys are mapped at all. For non-touch Android devices (e.g. I have an OUYA and Nexus Player) it's impossible without touch or a keyboard.

Would be helpful to have default keys mapped (e.g. Android knows how to deal with D-Pad and other keys without needing a specific keymap) to at least navigate the menu.

**Comment by andybotting** [comment]
Fixed with PR #227
